docs/legacy-project-tips.md

```diff
@@ -63,6 +63,44 @@ linting process more manageable, we're going to provide a
 some tips of tricks to break things down into bite size
 chunks.
 
+Pre-commit hook to lint only changed files
+------------------------------------------
+
+Using npm-scripts, some bash-fu and the [husky][husky] 
+module, you can have lint-trap only lint changed *.js files
+as a pre-commit hook. To do this, first install husky as a
+dev dependency:
+
+    $ npm install --save-dev husky
+
+Then add the following to your `package.json` file:
+
+    "scripts": {
+        ...
+        "lint-changed": "lint-trap `git diff HEAD --name-only --diff-filter=ACMRTU | awk '/\.js$/' | awk  -v path=$(git rev-parse --show-toplevel)/ '{print path$1}'`",
+        "precommit": "npm run lint-changed",
+        ...
+    },
+
+Husky will install git-hooks, including a pre-commit hook
```

Comment:
The reason I went with husky is because:
- it doesn't add any new fields to the package.json and instead just uses the scripts property
- it has a pre-push commit hook which you previously mentioned would be a better place to enforce certain checks since pre-commit might mess with the commit workflow of some developers. e.g. someone might prefer to only lint before pushing instead of on every commit.

I think having pre-push is pretty valuable since it's the safest place where we can enforce certain checks happening on all projects without messing with the workflow of individual developers.